docs/source/entities-advanced.mdx

```diff
@@ -902,3 +902,118 @@ const resolvers = {
+
+```graphql title="Example: using @context and @fromContext"
+scalar CurrencyCode;
+
+type Currency {
+   id: ID!
+   isoCode: CurrencyCode!
+}
+
+type User @key(fields: "id") @context(name: "userContext") {
+   id: ID!
+   lastTransaction: Transaction!
+   userCurrency: Currency!
+}
+
+type Transaction @key(fields: "id") {
+  id: ID!
+  currency: Currency!
+  amount: Int!
+  amountInUserCurrency(
+    currencyCode: CurrencyCode! 
+        @fromContext(field: "$userContext { userCurrency { isoCode } }")
+  ): Int!
+}
+```
+
+### Using multiple contexts 
+
+For an example using multiple contexts, we have a subgraph with three entities using two contexts, and a nested child entity referencing fields using those contexts: 
+
+```graphql title="Example: using multiple contexts"
+type Query {
+  a: A!
+  b: B!
+  c: C!
+}
+
+type A @key(fields: "id") @context(name: "context_1") @context(name: "context_2") {
+  id: ID!
+  field: String!
+  child: Child!
+} 
+
+type B @key(fields: "id") @context(name: "context_1") @context(name: "context_2") {
+  id: ID!
+  field: String!
+  child: Child!
+} 
+
+type C @key(fields: "id") @context(name: "context_1") {
+  id: ID!
+  anotherField: String!
+  child: Child!
+} 
+
+type Child @key(fields: "id") {
+  id: ID!
+  prop1(
+    arg: String! @fromContext(field: "$context_1 { field }")
+  ): Int!
+  prop2(
+    arg: String! 
+        @fromContext(field: "$context_2 ... A { field } ... B { field } ... C { anotherField }")
+  ): Int!
+}
+```
+
+When the same contextual value is set in multiple places—as is done in the example with the args of `Child.prop1` and `Child.prop2`—the `FieldValue` must resolve all types from each place into a single value that matches the parameter type.
+
+### Disambiguating contexts
+
+When a set context could be fulfilled by multiple ancestor entities in the type hierarchy, the nearest ancestor is chosen. For example, if both the parent and grandparent of a type can provide the value of a context, the parent is chosen because it's the closer ancestor.
+
+In the following example, given nested types `A`, `B`, and `C`, and `C` referencing a context that could be provided by either `A` or `B`, the value used by `C` is from `B` because it's a closer ancestor to `C` than `A`:
+
+```graphql
+type Query { 
+  a: A!
+}
+
+type A @key(fields: "id") @context(name: "context_1") {
+  id: ID!
+  field: String!
+  b: B!
+} 
+
+type B @key(fields: "id") @context(name: "context_1") {
+  id: ID!
+  field: String!
+  c: C!
+} 
+
+type C @key(fields: "id") {
+  id: ID!
+  prop(
+    arg: String! @fromContext(field: "$context_1 { field }")
+  ): Int!
+} 
+```
+
+In a more complex graph, a field could be reachable via multiple paths, and there could be a different field used to resolve the `prop` depending on which path was used.
```

Comment:
```suggestion
In a more complex graph, a field could be reachable via multiple paths, and there could be a different field used to resolve the `prop` depending on which path was used.

### Referencing fields across subgraphs

The definition of context scopes can only exist in one subgraph schema. You can not define `@context` in one subgraph and use `@fromContext` in another. However, you can use contexts to share data across subgraphs using the `@external` reference.

Using our first example, let's say we had a subgraph responsible for the `User` and `Currency` type

```graphql title="User subgraph"
scalar CurrencyCode

type Currency @shareable {
   id: ID!
   isoCode: CurrencyCode!
}

type User @key(fields: "id") {
   id: ID!
   userCurrency: Currency!
}
```

Then if in another subgraph you want to reference those fields, you can use the `@external` directive to still pass data across subgraph boundaries.

```graphql title="Transactions subgraph"
scalar CurrencyCode

type Currency @shareable {
   id: ID!
   isoCode: CurrencyCode!
}

type User @key(fields: "id") @context(name: "userContext") {
   id: ID!
   
  # This is a reference to the field resolved elsewhere
   userCurrency: Currency! @external
   
   # We add this field to our type here
   lastTransaction: Transaction! 
}

type Transaction @key(fields: "id") {
  id: ID!
  currency: Currency!
  amount: Int!
  amountInUserCurrency(
    currencyCode: CurrencyCode! 
        @fromContext(field: "$userContext { userCurrency { isoCode } }")
  ): Int!
}
```